File: src/bmsx/render/vdp/context_state.ts
import type { VDP } from '../../machine/devices/vdp/vdp';
import type { GameView } from '../gameview';

export function restoreVdpContextState(vdp: VDP, view: GameView): void {
	view.vdpFrameBufferTextures.initialize(vdp);
	view.vdpSlotTextures.initialize(vdp);
}

File: machine/cpp/src/render/vdp/context_state.cpp
#include "render/vdp/context_state.h"

#include "render/gameview.h"
#include "render/vdp/framebuffer.h"
#include "render/vdp/slot_textures.h"

namespace bmsx {

void restoreVdpContextState(VDP& vdp, GameView& view) {
	view.vdpFrameBufferTextures().initialize(vdp);
	view.vdpSlotTextures().initialize(vdp);
}

} // namespace bmsx
